### PR TITLE
[Fix] Support dynamic topk when num_classes < 5

### LINKED ISSUE #810: [Bug] RunTimeError: selected index k out of range
### 描述
An error will be reported when the 2 classification （<5）

```
RunTimeError: selected index k out of range
```

### 附加内容
修改处：

mmclassification\mmcls\models\losses\accuracy.py
Line:57
```
    # if num_classes < 5, will error
    maxk = min(max(topk), pred.size(1))

```

### PR DESCRIPTION
**Pull request by jamiechoi1995**
Fix #810, #667

When num_classes < 5, it will raise

```python
RunTimeError: selected index k out of range
```

One possible solution is to specify the topk value in config file:
```python
evaluation = dict(interval=1, metric='accuracy', metric_options={'topk': 1})
```

However, this PR propose a better solution, which is to calculate the topk value dynamically.

### CODE REVIEW

**Comment by Ezra-Yu** [comment]
Please sign the CLA, so that we can review this PR.

**Comment by jamiechoi1995** [comment]
> Please sign the CLA, so that we can review this PR.

Signed, thanks for the reminder.

**Review by mzr1996 — APPROVED** [review]
Looks good

File: mmcls/datasets/base_dataset.py [hunk 1]
```diff
@@ -145,8 +145,9 @@ def evaluate(self,
         Returns:
             dict: evaluation results
         """
+        maxk = min(5, len(results[0]))
```
Comment (mzr1996) [inline mmcls/datasets/base_dataset.py:148]: maybe directly use
```suggestion
        default_topk = (1, 5) if len(results[0]) > 5 else (1, )
```

File: mmcls/datasets/base_dataset.py [hunk 2]
```diff
@@ -145,8 +145,9 @@ def evaluate(self,
...
         if metric_options is None:
-            metric_options = {'topk': (1, 5)}
+            metric_options = {'topk': (1, maxk)}
```
Comment (mzr1996) [inline mmcls/datasets/base_dataset.py:150]: And here, use
```suggestion
            metric_options = {'topk': default_topk}
```

**Review by mzr1996 — CHANGES_REQUESTED** [review]
Hello, I think when `num_classes` less than 5, we can directly set the default topk as `(1, )` instead of `(1, maxk)`, because the topk-maxk result should be always 100%

**Comment by jamiechoi1995** [comment]
> Hello, I think when `num_classes` less than 5, we can directly set the default topk as `(1, )` instead of `(1, maxk)`, because the topk-maxk result should be always 100%

looks good to me.

**Comment by jamiechoi1995** [comment]
I have passed all the unit tests on my local machine and I have no idea why the online unit test failed, would anyone be able to help me take a look at this? @mzr1996 
图片

**Pull request closed without merging**